### ISSUE TITLE
[BUG] read_text doesn't apply a default

### ISSUE THREAD

**Issue by wence-** [rapidsai/cudf#11804]
**Describe the bug**

`read_text` advertisers `delimiter=None` (with perhaps some presumption that a default delimiter is put in place). In practice, no such thing occurs.

**Steps/Code to reproduce bug**

```python
import cudf
from io import StringIO

series = cudf.read_text(StringIO("foo bar"))

=> File text.pyx:33, in cudf._lib.text.read_text()

AttributeError: 'NoneType' object has no attribute 'encode'
```

**Expected behavior**

Should not get an `AttributeError`. Either a sensible default should be supplied, or the `delimiter` argument should be mandatory (and checked for `str`-iness).

**Comment by GregoryKimball** [comment]
@upsj I would be happy to either use `\n` default value or make `delimiter` a required argument. Please let me know if this turns out to be tricky.

**Comment by upsj** [comment]
I ran into the same issue and included a fix in #11802, i.e. making it a required argument
